core/src/main/java/io/grpc/Deadline.java

```diff
@@ -114,18 +114,15 @@ public Deadline offset(long offset, TimeUnit units) {
   /**
    * How much time is remaining in the specified time unit. Internal units are maintained as
    * nanoseconds and conversions are subject to the constraints documented for
-   * {@link TimeUnit#convert}. If there is no time remaining {@code 0} is always returned.
+   * {@link TimeUnit#convert}. If there is no time remaining, a negative offset from the
```

Comment:
Mentioning "system clock" is a bit strange, although it could be fine. You might consider something like, "If there is no time remaining, the returned duration is how long ago the deadline expired."